src/core/public/chrome/nav_group/nav_group_service.ts

```diff
@@ -114,12 +123,86 @@ export class ChromeNavGroupService {
   }
 
   private sortNavGroupNavLinks(
-    navGroup: NavGroupItemInMap,
+    navLinks: NavGroupItemInMap['navLinks'],
     allValidNavLinks: Array<Readonly<ChromeNavLink>>
   ) {
-    return getSortedNavLinks(
-      fulfillRegistrationLinksToChromeNavLinks(navGroup.navLinks, allValidNavLinks)
-    );
+    return getSortedNavLinks(fulfillRegistrationLinksToChromeNavLinks(navLinks, allValidNavLinks));
+  }
+
+  private getNavLinksForAllUseCase(
+    navGroupsMap: Record<string, NavGroupItemInMap>,
+    navLinks: Array<Readonly<ChromeNavLink>>
+  ) {
+    // Note: we need to use a new pointer when `assign navGroupsMap[ALL_USE_CASE_ID]?.navLinks`
+    // because we will mutate the array directly in the following code.
+    const navLinksResult: ChromeRegistrationNavLink[] = [
+      ...(navGroupsMap[ALL_USE_CASE_ID]?.navLinks || []),
+    ];
+
+    // Append all the links that do not have use case info to keep backward compatible
+    const linkIdsWithNavGroupInfo = Object.values(navGroupsMap).reduce((total, navGroup) => {
```

Comment:
`total` can confuse developers who work on this as it indirectly means a number. The norm for naming this is `accumulator` or `acc` (not as good).